src/cart/crt_hg.h

```diff
@@ -189,36 +189,8 @@ void crt_provider_get_ctx_list_and_num(bool primary, crt_provider_t provider, d_
 char* crt_provider_iface_str_get(bool primary, crt_provider_t provider, int iface_idx);
 struct crt_na_config*
 crt_provider_get_na_config(bool primary, crt_provider_t provider);
-
-static inline int
-crt_hgret_2_der(int hg_ret)
-{
-	switch (hg_ret) {
-	case HG_SUCCESS:
-		return 0;
-	case HG_TIMEOUT:
-		return -DER_TIMEDOUT;
-	case HG_INVALID_ARG:
-		return -DER_INVAL;
-	case HG_MSGSIZE:
-	case HG_OVERFLOW:
-		return -DER_OVERFLOW;
-	case HG_NOMEM:
-		return -DER_NOMEM;
-	case HG_CANCELED:
-		return -DER_CANCELED;
-	case HG_BUSY:
-		return -DER_BUSY;
-	case HG_FAULT:
-	case HG_PROTOCOL_ERROR:
-		return -DER_HG_FATAL;
-	case HG_PERMISSION:
-	case HG_ACCESS:
-		return -DER_NO_PERM;
-	default:
-		return -DER_HG;
-	};
-}
+int
+crt_hgret_2_der(int hg_ret);
```

Comment:
what is the reason for making that one non inline and leave the next one inline ?

Reply:
This is because we want daos to compile even with old version of mercury where `HG_HOSTFIREWALL` is not defined. Therefore, we enhanced scons to detect it and define one if it's not defined: https://github.com/daos-stack/daos/pull/16429/files#diff-433ec396346afa932f0d8f67525073e133a0e3dbb92b15f9503e81574a2e3e36R113.

The reason to make it non-inline is to move it out of header file. In that case, we can limit the definition above limited to files under cart directory. Otherwise, `-DHG_HOSTFIREWALL=0xdeadbeaf` has to be applied all files where `crg_hg.h` is used.